### PR TITLE
Filter screen data v3

### PR DESCRIPTION
**Pull request by mitchdowney**

### CODE REVIEW

File: src/screens/FilterScreen.tsx [hunk 1]
```diff
@@ -69,84 +126,106 @@ export class FilterScreen extends React.Component<Props, State> {
...
+      sections
+    } = generateSections(options)
+
+    return {
+      selectedCategoryItemKey: newSelectedCategoryItemKey,
+      selectedCategorySubItemKey: newSelectedCategorySubItemKey,
+      selectedFilterItemKey: newSelectedFilterItemKey,
+      selectedSortItemKey: newSelectedSortItemKey,
+      sections,
+      stateKey
```
Comment (kreonjr) [inline src/screens/FilterScreen.tsx:213]: If none of your "ifs" get caught here, stateKey will be set to undefined. Is that expected?

File: src/services/category.ts [hunk 2]
```diff
@@ -27,3 +27,46 @@ export const downloadCategoriesList = async () => {
...
+        return {
+          label: category.title,
+          value: category.id,
+          ...category
+        }
+      })
+    }
+    return categoryItems
+  } catch (err) {
+    console.log('Bottom Selection Bar error: ', err)
```
Comment (kreonjr) [inline src/services/category.ts:59]: Do you want to pull out `let categoryItems = []` out of the try catch and then return it at the very end, after the catch? This way if the function throws, it won't return undefined

File: src/services/category.ts [hunk 3]
```diff
@@ -27,3 +27,46 @@ export const downloadCategoriesList = async () => {
...
+
+export const getFlatCategoryItems = async () => {
+  const categoryItems = await getCategoryItems()
+  return generateFlatCategoryItems(categoryItems)
+}
+
+export const getDefaultCategory = async () => {
+  const flatCategoryItems = await getFlatCategoryItems()
+  const defaultItem = flatCategoryItems[0]
+  return defaultItem
```
Comment (kreonjr) [inline src/services/category.ts:71]: This will be undefined if `flatCategoryItems` is empty. Should it fallback to an empty object?

**Review by kreonjr — CHANGES_REQUESTED** [review]

File: src/screens/FilterScreen.tsx (hunk 1 above)
Comment (mitchdowney) [reply src/screens/FilterScreen.tsx:213]: Looks like stateKey isn't even used anywhere 😬

**Pull request merged**